src/SurfacePlot/SurfacePlot.tsx

```diff
@@ -0,0 +1,202 @@
+            scene: {
+              xaxis: {
+                color: theme.palette.mode === "light" ? "" : "white",
+                gridcolor:
+                  theme.palette.mode === "light"
+                    ? ""
+                    : theme.palette.grey["500"],
+                showgrid: false,
+                title: {
+                  font: {
+                    family: "Montserrat",
+                    size: 12
+                  },
+                  text: xlabel || ""
+                }
+              },
+              yaxis: {
+                color: theme.palette.mode === "light" ? "" : "white",
+                gridcolor:
+                  theme.palette.mode === "light"
+                    ? ""
+                    : theme.palette.grey["500"],
+                title: {
+                  font: {
+                    family: "Montserrat",
+                    size: 12
+                  },
+                  text: ylabel || ""
+                }
+              },
+              zaxis: {
+                color: theme.palette.mode === "light" ? "" : "white",
+                gridcolor:
+                  theme.palette.mode === "light"
+                    ? ""
+                    : theme.palette.grey["500"],
+                title: {
+                  font: {
+                    family: "Montserrat",
+                    size: 12
+                  },
+                  text: zlabel || ""
+                }
+              }
+            }
+          }}
+          style={{ flexGrow: 1, height: "100%", minHeight: 0, width: "100%" }}
+          useResizeHandler={true}
+          config={config}
+        />
+      </Box>
+    </ConditionalDialog>
+  );
+};
+
+export default SurfacePlot;
+
+// svg path for fullscreen icon in plotly menu bar
+const fullscreenIcon = {
+  height: 1792,
+  path: "M256 1408h1280v-768h-1280v768zm1536-1120v1216q0 66-47 113t-113 47h-1472q-66 0-113-47t-47-113v-1216q0-66 47-113t113-47h1472q66 0 113 47t47 113z",
+  width: 1792
+};
+
+type ConfigProps = {
+  isFullscreen: boolean;
+  handleClickFullscreen: () => void;
+};
+
+/**
+ * Returns a plotly config function with provided callback for clicking fullscreen
+ */
+const getConfig = ({ isFullscreen, handleClickFullscreen }: ConfigProps) => {
+  return {
+    displaylogo: false, // never display plotly logo
+    modeBarButtonsToAdd: !isFullscreen // if we are not in full screen, show a button to launch to fullscreen
+      ? [
+          {
+            click: handleClickFullscreen,
+            direction: "up",
+            icon: fullscreenIcon,
+            name: "Fullscreen",
+            title: "Fullscreen"
+          }
+        ]
+      : []
+  };
+};
+
+type ConditionalDialogProps = {
+  condition: boolean;
+  onClose: () => void;
+  children: React.ReactNode;
+  dialogTitle?: string;
+};
+
+/**
+ * Returns children in a fullscreen dialog if condition is true, otherwise just returns children
+ */
+function ConditionalDialog({
```

Comment:
There seems to be a lot of wasted white space around the plot component, particularly so in full screen mode. The plot is very small here, I can't read the axis labels and there is a lot of white space. Please discuss with @Sowbhagya-ipg regarding the size of the plot and margins etc.

![image](https://github.com/IPG-Automotive-UK/react-ui/assets/8976377/bc161019-95d9-4e63-b19c-fccfdbe683f4)

Reply:
Was unable to remove the white space easily, but I have made improvement to make it larger.